### PR TITLE
Fixed #28670 -- Added FETCH/OFFSET support on Oracle.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28670

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
`queries.tests.SubqueryTests.test_distinct_ordered_sliced_subquery` should failed, because SQL is invalid on Oracle. I prepared separate PR with fix #9207.

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/models/sql/compiler.py [hunk 1]
```diff
@@ -439,6 +439,9 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
         try:
             extra_select, order_by, group_by = self.pre_sql_setup()
             for_update_part = None
+            # Indicates whether we need to construct the SQL needed to use
```
Comment (timgraham) [inline django/db/models/sql/compiler.py]: Maybe: `# Is a LIMIT/OFFSET clause needed?`

(Avoid use of "we" in comments per our [style guide](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).)

File: django/db/backends/oracle/operations.py [hunk 2]
```diff
@@ -233,8 +231,16 @@ def field_cast_sql(self, db_type, internal_type):
         else:
             return "%s"
 
+    def no_limit_value(self):
+        return None
+
     def limit_offset_sql(self, low_mark, high_mark):
-        return ''
+        """Return FETCH/OFFSET SQL clause."""
```
Comment (timgraham) [inline django/db/backends/oracle/operations.py]: Usually we don't repeat the docstring from the base class.

**Comment by felixxm** [comment]
@timgraham Thanks for review.

**Pull request merged**